Registration Selection Page

miguelaenlle opened the pull request:
- Implemented the registration selection page designs
- Extracted landing page "Register Now" button with shine animation into the new `OlympianButton` component, and reused that in the registration selection page
- Optimized for laptop, desktop, and mobile screen formats (ignore the yellow guidelines that appeared from Inspect Element):

Laptop:
<img width="1470" alt="Screenshot 2024-12-29 at 1 54 58 PM" src="https://github.com/user-attachments/assets/d30fa4d6-b890-4c16-ae96-dc56bfe0613f" />

Desktop:
![ ](https://github.com/user-attachments/assets/9d9f5250-56fc-4355-9348-2f463ef43895)

Mobile:
<img width="341" alt="Screenshot 2024-12-29 at 1 55 15 PM" src="https://github.com/user-attachments/assets/a247c248-4b8e-42a8-8984-ecb2371bc458" />

- Moved registration question page to `/register/general` instead of keeping it at `/register`

miguelaenlle committed:
Implemented Track Selection page

miguelaenlle committed:
Improved mobile/desktop optimization

miguelaenlle committed:
Re-enable general registration redirect

miguelaenlle committed:
Set pro track challenge success/failure screens to redirect to the general signup

Timothy-Gonzalez commented on app/register/page.tsx:
```diff
@@ -1,43 +1,35 @@
                     window.location.replace("/profile");
                 }
-
                 if ("_id" in registration) {
```
This shouldn't exist (`_id` is not going to be sent in the future once API is fixed, and should not be used as a "registration from database" marker) - I realize you didn't add this but this should be removed - see main review.

miguelaenlle replied:
Got it. I removed this code.

Timothy-Gonzalez reviewed (changes requested):
Looks great! I think for mobile image you show the snowglobe can get a bit smaller, since we want the globe to be in proportion to the logo. Probably just tweaking the sizing a bit.  In chrome's device emulator, `iPhone 12 Pro` is a good min-size test for phone ratios:
![image](https://github.com/user-attachments/assets/1216ac6c-31c7-4264-8aaa-ddc2ed3897cc)


For the redirects, it looks like we redirect to `/register/general` if they've already started registration, but I think we should stay at the selection page instead. The logic of determining if they've already started is very cursed and makes it hard to return to the challenge. For example, the back button currently does not work and if you start registration you cannot return to the challenge at all. I think auto-redirects should just be removed, so this means removing the hasChosen/setHasChosen stuff as well. If I go to `/register/`, I should be able to select what I want instead of being sent wherever.

Finally, there's a small issue with the button where it shows as being hovered over, but the link isn't clickable (see below). Ideally the a tag should wrap the button and have all hover events (click, move down, etc.) be tied to the a tag so they are in sync. Otherwise you have sync issues where it looks like you're hovering over it but clicking does nothing (if you hover over the padding or border of the button instead of the text).
![image](https://github.com/user-attachments/assets/880fd65c-b708-45b5-923a-6c4f0dd68aa1)
*For example, clicking the above does nothing since the link is the text, ideally links should wrap their clickable element*

Overall these are pretty minor things and most of it wasn't from your change, I just want these issues to be fleshed out as well. Thanks!

miguelaenlle committed:
Removed track selection redirect logic; improved iPhone mobile optimization; fixed button hover issue

miguelaenlle commented:
- Shrunk snowglobe on mobile:

![Screenshot 2024-12-30 at 11 06 39 AM](https://github.com/user-attachments/assets/bf19908e-f436-4079-baae-caeeec2c7231)

- Fixed the OlympianButton link issue

- Removed redirect logic from the /register page

miguelaenlle committed:
Added redirect to back button of RegistrationForm

miguelaenlle committed:
Added redirect from /register page; added backwards saving

miguelaenlle committed:
Moved reimbursement question; removed how will you get to ... question; changed travel acknowledgement validation

miguelaenlle commented:
- Moved `Would you like to be considered for travel reimbursement` question to transportation page:
![image](https://github.com/user-attachments/assets/7e26e189-a601-4465-b119-075b953a516b)

- Removed `How will you be getting to HackIllinois?` question

- Defaulted `travelAcknowledge` to not filled in; required that the user clicks "Yes"
![image](https://github.com/user-attachments/assets/d90ea49e-707a-4954-9aad-d40fac23fd67)

miguelaenlle committed:
Required that travel acknowledgement has an option selected

Timothy-Gonzalez commented on components/Registration/Pages/Transportation/Transportation.tsx:
```diff
@@ -17,15 +17,15 @@ const travelAcknowledgeOptions = [
 ] satisfies CheckboxOption[];
 
-const travelMethodOptions = [
+const travelReimbursementOptions = [
```
Make the only option "I acknowledge" (value can be YES if it makes it easier)

Timothy-Gonzalez reviewed (changes requested):
Tested everything - everything works as expected! One minor change after seeing the yes/no - we shouldn't have a no option, just a "I acknowledge". I promise this is the final change lol.

miguelaenlle committed:
Implemented are you aware? question

miguelaenlle committed:
Removed type=submit

miguelaenlle committed:
Removed type=submit change

Timothy-Gonzalez committed:
Revert saving on previous

The form rework required for this will be done in another branch.
This pr has overgrown it's stay.
